pkg/provenance/sign_test.go

```diff
@@ -266,6 +270,85 @@ func TestClearSign(t *testing.T) {
 	}
 }
 
+func TestMixedKeyringRSASigningAndVerification(t *testing.T) {
+	signer, err := NewFromFiles(testKeyfile, testMixedKeyring)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(signer.KeyRing) == 0 {
+		t.Fatal("expected signer keyring to be loaded")
+	}
+
+	hasEdDSA := false
+	for _, entity := range signer.KeyRing {
+		if entity.PrimaryKey != nil && entity.PrimaryKey.PubKeyAlgo == packet.PubKeyAlgoEdDSA {
+			hasEdDSA = true
+			break
+		}
+
+		for _, subkey := range entity.Subkeys {
+			if subkey.PublicKey != nil && subkey.PublicKey.PubKeyAlgo == packet.PubKeyAlgoEdDSA {
+				hasEdDSA = true
+				break
+			}
+		}
+
+		if hasEdDSA {
+			break
+		}
+	}
```

Comment:
I think this could have been written something like (for future reference):
```
hasEdDSA := slices.ContainsFunc(signer.KeyRing, func(entity Entity) {
   if entity.PrimaryKey != nil && entity.PrimaryKey.PubKeyAlgo == packet.PubKeyAlgoEdDSA) {
     return true
   }
     
   return slices.Contains(entity.Subkeys, func(subkey SubKey) {
     return subkey.PublicKey != nil && subkey.PublicKey.PubKeyAlgo == packet.PubKeyAlgoEdDSA
   })
})
```

(a lot more compact and tidy IMHO)